module/VuFind/src/VuFind/Controller/MyResearchController.php

```diff
@@ -877,12 +879,9 @@ public function deleteAction()
 
         // If we got this far, the operation has not been confirmed yet; show
         // the necessary dialog box:
-        if (empty($listID)) {
-            $list = false;
-        } else {
-            $table = $this->getTable('UserList');
-            $list = $table->getExisting($listID);
-        }
+        $list = empty($listID)
+            ? false
+            : $this->getDbService(UserListServiceInterface::class)->getUserListById($listID);
```

Comment:
@aleksip, your impressions are correct, but there are some practical reasons why some early bad design decisions in our controllers have proven difficult to eliminate.

Cleaning up this mess has been on the wishlist for a decade now -- see [VUFIND-966](https://openlibraryfoundation.atlassian.net/browse/VUFIND-966). I've just posted some comments on that ticket including an explanation of why it has been so slow to move, and how I think we might be able to make progress on it. I'd be happy to continue this conversation over there, which I think is probably the best place to capture thoughts.

For the moment, I think it's probably best to leave this as it is, since we're following an existing pattern, and this is not the right time to try to fix this. But in the long term, I absolutely agree that this is not good, and I look forward to having some bandwidth to really fix it properly. For now, though, this database refactoring has to get done first!

Reply:
However, I _thought_ this comment was on my controller refactoring PR, whereas it was actually somewhere different. So apologies for responding in entirely the wrong context. My other thoughts about controllers still stand, but let me see if I can also make this cleaner here. :-)

Reply:
...and now I see that even though this wasn't the PR I thought it was, the comments did still apply to the controller. So no need to apologize. It's still early, and I have a lot of reviews to sift through. I'm sure my brain will start working right in a few more minutes. ;-)